packages/components/src/internal/constants.ts

```diff
@@ -49,85 +49,97 @@ export enum EXPORT_TYPES {
     LABEL_TEMPLATE,
 }
 
+export enum EDIT_METHOD {
+    BULK_EDIT = 'BulkEdit',
+    BULK_EDIT_LINEAGE = 'BulkEditLineage',
+    DETAIL_EDIT = 'DetailEdit',
+    DETAIL_EDIT_LINEAGE = 'DetailEditLineage',
+    FORM_INSERT = 'FormInsert',
+    GRID_EDIT = 'GridEdit',
+    GRID_INSERT = 'GridInsert',
+    STORAGE_VIEW_ACTION = 'StorageViewAction',
+}
+
 export enum KEYS {
-    Backspace = 8,
-    Tab = 9,
-    Enter = 13,
-    Shift = 16,
-    Ctrl = 17,
+    A = 65,
+    Add = 107,
     Alt = 18,
-    PauseBreak = 19,
-    CapsLock = 20,
-    Escape = 27,
-    Space = 32,
-    PageUp = 33,
-    PageDown = 34,
-    End = 35,
-    Home = 36,
-
-    LeftArrow = 37,
-    UpArrow = 38,
-    RightArrow = 39,
-    DownArrow = 40,
-
-    Insert = 45,
-    Delete = 46,
-
-    Zero = 48,
-    ClosedParen = Zero,
-    One = 49,
-    ExclamationMark = One,
-    Two = 50,
-    AtSign = Two,
-    Three = 51,
-    PoundSign = Three,
-    Hash = PoundSign,
-    Four = 52,
-    DollarSign = Four,
-    Five = 53,
-    PercentSign = Five,
-    Six = 54,
-    Caret = Six,
-    Hat = Caret,
     Seven = 55,
     Ampersand = Seven,
     Eight = 56,
     Star = Eight,
     Asterik = Star,
-    Nine = 57,
-    OpenParen = Nine,
-
-    A = 65,
+    Two = 50,
+    AtSign = Two,
     B = 66,
+    Backspace = 8,
     C = 67,
+    CapsLock = 20,
+
+    Six = 54,
+    Caret = Six,
+    ClosedBracket = 221,
+    Zero = 48,
+
+    ClosedParen = Zero,
+    Comma = 188,
+
+    Ctrl = 17,
     D = 68,
+    Dash = 189,
+    DecimalPoint = 110,
+    Delete = 46,
+    Divide = 111,
+    Four = 52,
+    DollarSign = Four,
+    DownArrow = 40,
     E = 69,
+    End = 35,
+    Enter = 13,
+    Equals = 187,
+    Escape = 27,
+    One = 49,
+    ExclamationMark = One,
     F = 70,
+    F1 = 112,
+    F2 = 113,
+    F3 = 114,
+    F4 = 115,
+    F5 = 116,
+    F6 = 117,
+
+    F7 = 118,
+    F8 = 119,
+    F9 = 120,
+    F10 = 121,
+    F11 = 122,
+    F12 = 123,
+    FFLeftMetaKey = 224, // Firefox
+    Five = 53,
+    ForwardSlash = 191,
     G = 71,
+    Tilde = 192,
+    GraveAccent = Tilde,
     H = 72,
+    Three = 51,
+    PoundSign = Three,
+    Hash = PoundSign,
+    Hat = Caret,
+    Home = 36,
     I = 73,
+    Insert = 45,
     J = 74,
     K = 75,
     L = 76,
+    LeftArrow = 37,
+    LeftMetaKey = 91,
     M = 77,
-    N = 78,
-    O = 79,
-    P = 80,
-    Q = 81,
-    R = 82,
-    S = 83,
-    T = 84,
-    U = 85,
-    V = 86,
-    W = 87,
-    X = 88,
-    Y = 89,
-    Z = 90,
```

Comment:
the linting on this object is terrible! I think it makes way more sense to have these letters together even though it means they are technically "out of order".